### PR TITLE
[14.0][MIG] purchase_order_type_advanced

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Standard migration to v14
--
`type `field in v13 is renamed in v14 to `move_type` and
`.with_context(force_company=...)` became `.with_company(...)`

### CODE REVIEW

**Comment by ChristianSantamaria** [comment]
@dalonsod, could you check it un Test environment? Thanks!

File: purchase_order_type_advanced/__manifest__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
     """,
     "author": "Solvos",
     "license": "LGPL-3",
-    "version": "13.0.1.0.2",
+    "version": "14.0.1.0.2",
```
Comment (dalonsod) [inline purchase_order_type_advanced/__manifest__.py]: Reset version to `14.0.1.0.0`

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comment.
When testing, I've found an error, but not sure if this addon is involved:

- Created a second purchase order sequence for VGO Purchases.
- Created a second warehouse "VG".
- Created a new purchases journal TEMP.
- Created a new purchase order type for VGO purchases... with the above sequence, the incoming operation type for VG as default and default journal, the TEMP one.
- When confirmed a VGO purchase and validated the picking, product were placed in VGO stock location (OK).
- But an error is raised when I've tried to validate that the created invoiced assigned to the proper journal:

imagen

could you check?

cc @lmiguens-solvos

File: purchase_order_type_advanced/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                order.picking_type_id = order.order_type.picking_type_id
+
+    def action_view_invoice(self):
+        res = super(PurchaseOrder, self).action_view_invoice()
+        ctx = {}
+        if self.order_type.journal_id:
+            ctx["default_journal_id"] = self.order_type.journal_id.id
+        if self.order_type:
+            ctx["default_purchase_type_id"] = self.order_type.id
+        res["context"].update(ctx)
```
Comment (dalonsod) [inline purchase_order_type_advanced/models/purchase_order.py]: In v14 this `res["context"]` seems to be a string, not a dictionary, so this `update()` fails.
Tested in AR Test Environment, clicking in purchase invoices button, this error is raised:

```
  File "/home/odoo/odoo-test/erp-ariveira/src/slv-purchase/purchase_order_type_advanced/models/purchase_order.py", line 24, in action_view_invoice
    res["context"].update(ctx)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/odoo/odoo-test/erp-ariveira/src/odoo/odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/odoo-test/erp-ariveira/src/odoo/odoo/http.py", line 316, in _handle_exception
    raise exception.with_traceback(None) from new_cause
AttributeError: 'str' object has no attribute 'update'
```

Comment (dalonsod) [reply purchase_order_type_advanced/models/purchase_order.py]: Solved!

**Comment by dalonsod** [comment]
> * But an error is raised when I've tried to validate that the created invoiced assigned to the proper journal

Solved by the last commit

**Comment by dalonsod** [comment]
@ChristianSantamaria please update in both customers Test environments, so we'll test it again. Thanks!

**Comment by dalonsod** [comment]
Last commit is still untested, so I've move this PR back to draft

**Comment by dalonsod** [comment]
Already validated by at least one customer, let's merge it...

**Pull request merged**